mailing/templates.go

```diff
@@ -0,0 +1,39 @@
+package mailing
+
+import (
+	"context"
+	"errors"
+)
+
+// ErrTemplateNotFound is returned when a given template is not found inside the list of available templates.
+var ErrTemplateNotFound = errors.New("template not found")
+
+// templateWrapper implements Sender allowing the template management to be handled by an inner map between
+// a key and a template location.
+type templateWrapper struct {
+	templates map[string]string
+	sender    Sender
+}
+
+// Send validates that the given template is contained inside the list of available templates and sends the email.
+func (t *templateWrapper) Send(ctx context.Context, sender string, recipients, cc, bcc []string, subject, template string, data any) error {
+	value, ok := t.templates[template]
+	if !ok {
+		return ErrTemplateNotFound
+	}
+	return t.sender.Send(ctx, sender, recipients, cc, bcc, subject, value, data)
+}
+
+// NewTemplateWrapper initializes a new Sender implementation that conveniently handles templates by configuring
+// the different available email templates in the given templates argument. It relies on the given sender to actually
+// send the email. This wrapper can be used alongside the AWS SES and Sendgrid implementations.
```

Comment:
Note that this may change if we decide to change the name of `TemplateWrapper` to `TemplateSender`.

```suggestion
// NewTemplateWrapper initializes a new Sender implementation that sends pre-defined templates
// using another email sender. Emails are sent by specifying the template to use and providing data
// for them.
// `sender` contains a Sender implementation. This is what is actually used to send emails. 
// `templates` maps template identifiers to template filepaths. The identifiers are used to select
// the template to send.
```